functions-and-scope.js

```diff
@@ -1,83 +1,161 @@
+
+// Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde cijfer
+// teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of
+// [8, 9, 4, 6, 10] Log het antwoord in de terminal.
+
+function averageGrade(Array) {
+    let sum = 0;
+    for (let i = 0; i < Array.length; i++) {
+        sum += Array[i];
+    }
+    return sum / Array.length;
+}
+console.log(averageGrade(grades));
+console.log(averageGrade([6, 4, 5]));
+console.log(averageGrade([8, 9, 4, 6, 10]) + "\n");     // "\n" voor opmaak output //
 
 // ---- Verwachte uitkomsten:
 // averageGrade(grades) geeft 6.642857142857143
-// averageGrade([6, 4, 5]) geeft xxxx
-// averageGrade([8, 9, 4, 6, 10]) geeft xxxx
+// averageGrade([6, 4, 5]) geeft 5
+// averageGrade([8, 9, 4, 6, 10]) geeft 7.4
+
 
 
 /* 2c: Afronden op twee decimalen */
+
 // Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
 // Tip: Google is your best friend!
 
-
+function twoDecimalGrade(Array) {
+    const decimal = averageGrade(Array)
+    return decimal.toFixed(2)
+}
+console.log(twoDecimalGrade(grades));
+console.log(twoDecimalGrade([6, 4, 5]));
+console.log(twoDecimalGrade([8, 9, 4, 6, 10]) + "\n");     // "\n" voor opmaak output //
 
 
 /* Bonusopdracht: hoogste cijfer */
 
 /* 3a: Script schrijven  */
-// Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
+
+// Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier
+// geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
 // * Hoe kan ik iedere waarde van de array langsgaan?
 // * Op welke conditie moet ik checken?
 // * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
 // Log het antwoord in de terminal.
 
+grades.sort();
+let last = grades[grades.length - 1];
+if (grades[0] === 10)
+    console.log(10);
+else
+    console.log(last + "\n");       // "\n" voor opmaak output //
+
 // ---- Verwachte uitkomst: 9
 
 
+
 /* 3b: Omschrijven tot een herbruikbare functie */
+
 // Schrijf een functie genaamd highestGrade, die een array van cijfers verwacht (zoals grades) en het hoogste cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 3a.
 // Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
 
+function highestGrade(Array) {
+    Array.sort();
+    let last = Array[Array.length - 1];
+    if (Array[0] === 10)
+        console.log(10);
+    else
+        console.log(last);
```

Comment:
Dit kan volgens mij nog iets simpeler
```suggestion
    Array.sort();
    let last = Array[Array.length - 1];
    console.log(last);
```